### ISSUE TITLE
Redesign the Data Flow and Component Structure

### ISSUE THREAD

**Issue by akshay2905** [cbgaindia/story-generator#64]
The app works fine, but there is lot of room improvement in the design of the app. Following are the major improvements that can be made with minimal efforts - 

- Redesign the data flow which is compatible with different levels of hierarchy.
- Based on the above design prepare suitable routes for the components. 
- Headers and Footers can be moved to MainViewComponent to reduce the code redundancy. Currently they are placed in each visualization component, thus giving rise to duplicacy.

**Comment by VenkateshKatabathula** [comment]
Hi @akshay2905, I've seen a pull request has been merged for this issue. Is it complete. Can u bring some light into this. I'm interested to contribute ☺

**Comment by akshay2905** [comment]
Hey @VenkateshKatabathula Great to hear from you. The pull request I had generated was part of a series of iteration that would be required to redesign the structure. Each iteration would have one component redesigned to the new structure. The first one was done to showcase the basic structure of how the new components would be designed.  I would recommend you to setup the project and explore it for a while. File bugs or possible enhancements you would like to recommend. I would add specific issues or feature request once a couple of iterations are completed, which should be hopefully the end of this week. Ping me if you face any issues in setting up the project.

**Comment by VenkateshKatabathula** [comment]
Thanks @akshay2905. I'll start contributing 😃

**Comment by VenkateshKatabathula** [comment]
just had a look at project. Just saw there were lot of lint and all. however I was not able to see anything on webpage. I am getting this error in console. please help :)
![console](https://cloud.githubusercontent.com/assets/10276910/26600110/7180cf2c-4598-11e7-9ff8-267ccfeae4c0.JPG)

**Comment by akshay2905** [comment]
@VenkateshKatabathula I have updated the process, as the version 2.0 development was executed in a different scenario. Please use the current develop branch to update the new changes. Sharing the link of the current process which has been adopted after the first release - [Branching Model](https://github.com/cbgaindia/story-generator/wiki/Branching-Model). I am currently working on redesign-component-sidebar branch which has redesign-app as the parent branch. Thus all the redesign-[component-name] would be merged into redesign-app. Once the redesign is complete we would merge it into develop and then master for production.

Yes there are linting issues, but they wont affect working of the project. Most of them are warnings, which we would need to look into. Sadly I wasnt able to find a design pattern or framework in which we can design our app into. Thus the redesigning is taking place. Most of the issues are related to that. 

The good thing about the React community is the release cycle. We can surely update the packages, but need to make sure that all the components are compatible with it. I did try to update it, but was facing compatibility issues with some of them. We would definitely look into it, once the redesign phase is complete.
